mkosi/qemu.py

```diff
@@ -702,12 +709,28 @@ def run_qemu(args: Args, config: Config, qemu_device_fds: Mapping[QemuDeviceNode
 
         for tree in config.runtime_trees:
             sock = stack.enter_context(start_virtiofsd(tree.source, uidmap=True))
+            tag = tree.target.name if tree.target else tree.source.name
             cmdline += [
                 "-chardev", f"socket,id={sock.name},path={sock}",
-                "-device", f"vhost-user-fs-pci,queue-size=1024,chardev={sock.name},tag={sock.name}",
+                "-device", f"vhost-user-fs-pci,queue-size=1024,chardev={sock.name},tag={tag}",
             ]
             target = Path("/root/src") / (tree.target or tree.source.name)
-            kcl += [f"systemd.mount-extra={sock.name}:{target}:virtiofs"]
+            kcl += [f"systemd.mount-extra={tag}:{target}:virtiofs"]
+
+        if want_scratch(config) or config.output_format in (OutputFormat.disk, OutputFormat.esp):
+            cmdline += ["-device", "virtio-scsi-pci,id=scsi"]
+
+        if want_scratch(config):
+            scratch = stack.enter_context(tempfile.NamedTemporaryFile(dir="/var/tmp", prefix="mkosi-scratch"))
+            scratch.truncate(1024**4)
```

Comment:
Thats a terabyte and while I think truncating the file to that size in tmpfs will not allocate a file this large, is this guaranteed for all FS or could this go haywire in some edge case and fill someone's space?

Reply:
We do the same thing in systemd-repart and nobody's complained yet, so I think it should be fine.